File: Fluffle.Utils/ShortUuidDateTime.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Noppes.Fluffle.Utils
{
    public static class ShortUuidDateTime
    {
        private const string Alphabet = "ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz23456789";
        private static IDictionary<char, int> AlphabetReverse = Alphabet
            .Select((c, i) => (c, i))
            .ToDictionary(x => x.c, x => x.i);

        private const int YearOffset = 2020;
        private const int MinuteMultiplier = 2;

        public static string ToString(DateTime value)
        {
            var year = Alphabet[value.Year - YearOffset];
            var month = Alphabet[value.Month];
            var day = Alphabet[value.Day];
            var hour = Alphabet[value.Hour];
            var minute = Alphabet[value.Minute / MinuteMultiplier];

            return $"{year}{month}{day}{hour}{minute}";
        }

        public static DateTime FromString(string value)
        {
            if (value.Length < 5)
                throw new ArgumentException($"A {nameof(ShortUuidDateTime)} consists out of at least 5 characters.", nameof(value));

            var year = AlphabetReverse[value[0]] + YearOffset;
            var month = AlphabetReverse[value[1]];
            var day = AlphabetReverse[value[2]];
            var hour = AlphabetReverse[value[3]];
            var minute = AlphabetReverse[value[4]] * MinuteMultiplier;

            return new DateTime(year, month, day, hour, minute, 0);
        }
    }
}

File: Fluffle.UI/src/services/short-uuid-date-time.ts
import { DateTime } from 'luxon'

export default class ShortUuidDateTime {
    static fromString(id: string) {
        let alphabet = {};
        Array.from('ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz23456789').forEach((char, i) => alphabet[char] = i);

        let year = alphabet[id[0]] + 2020;
        let month = alphabet[id[1]];
        let day = alphabet[id[2]];
        let hour = alphabet[id[3]];
        let minute = alphabet[id[4]] * 2;
        let dateTime = DateTime.fromFormat(`${year} ${month} ${day} ${hour} ${minute}`, 'y L d H m', { zone: 'utc' });

        return dateTime;
    }
}
